SubterfugeCore/Core/Generation/NameGenerator.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SubterfugeCore.Core.Timing;
+
+namespace SubterfugeCore.Core.Generation
+{
+    public class NameGenerator
+    {
+        private SeededRandom seeder;
+
+        List<string> _outpostNames;
+        
+        /// <summary>
+        /// Creates an instance of the name generator using an already generated seeded random tool.
+        /// </summary>
+        /// <param name="seeder">The seeder to use for random name selection</param>
+        public NameGenerator(SeededRandom seeder)
+        {
+            this.seeder = seeder;
+            this.populateNames();
+        }
+
+        /// <summary>
+        /// Sets the list of outpost names to be pulled from.
+        /// </summary>
+        public void populateNames()
+        {
+            _outpostNames = new List<string>() {
+            "Rokovo",
+            "Latvia",
+            "Pichu",  // Pokemon easer egg
+            "Shiloh", // My dog. RIP <3
+            "Dozer",
+            "Nautilus",
+            "Minceraft", // Pretty obvious
+            "Sans", // Undertale
+            "London",
+            "Helsinki",
+            "Subterfuge",
+            "Lyons",
+            "Norris",
+            "Hooper",
+            "Glover",
+            "Mills",
+            };
+        }
+
+        public string GetRandomName()
+        {
+            if (_outpostNames.Count > 0)
+            {
+                int selection = seeder.NextRand(0, _outpostNames.Count - 1);
+                string name = _outpostNames[selection];
+                _outpostNames.Remove(name);
+                return name;
+            }
+            // Should never get here but...
```

Comment:
It's actually quite plausible to get here if the game is configured to have more outposts than that there are names. Since these two numbers live in entirely different sections of code I think it's very good to have this kind of fallback logic :)

Reply:
Although (if you feel like it) it may be fun to make this randomiser a little smarter and let it generate for example only combinations of consonant+vowel syllables. That way the random string can still be pronounced and therefore appears more legible.

Reply:
There is also the astronomically small chance that this randomiser generates a name that is already in use (for example "Nautilus", as it has 8 letters)!

Reply:
Yeah, I agree that this 'catchall' is not totally unique. The reason the comment says `// Should never get here but...` is because I assume we will have a TON of outpost names and should never get to that location because of the abundance of outpost names. However, its good to have a fallback. I will see what I can do to ensure uniqueness for as long as possible once we run out of names as well as try to make the names a bit smarter

Reply:
I'm notoriously nitpicky sometimes when it comes to these things (both to the joy and frustration of my coworkers) so I understand as well if you decide to ignore this comment thread (and related comments). I'll leave it up to you if you can think of something cool and elegant 😄

Reply:
I added two new fallback mechanisms.
---
The first fallback takes the 1st and 3rd characters of an existing outpost name and swaps them with the ending of another outpost name.

For example, "Rokovo" and "Latvia" could make:
"Ratvia", "Rokvia", "Lokovo", and "Latovo"

This generates two names for each of the XP2 (X perm 2) permutations of default outpost names. We currently have 16 outposts so this method generated (2 * 16P2) 480 new outpost names as fallbacks (with the exception of possible duplicates) from just 16 initial outpost names. Scrolling through the list, the generated names are generally understandable and pronounceable as they were derived from words that already exist.

These names only get selected after the default set of outpost names has been exhausted.
---
The second fallback mechanism takes the letters in a random outpost and attempts to shuffle the letters (up to 10 times) looking for a unique combination. For example "rokovo" could be "ooorkv". This is pretty failsafe and in generating a list of 50,000 names, only about 10 got to the 2nd (of 10) scramble attempt.
---
The final fallback is the same as before, a randomly generated string of letters.